src/engine/Resources/Sound.ts

```diff
@@ -582,6 +607,25 @@ class WebAudioInstance implements IAudio {
 
       return this._playingPromise;
    }
+   
+   public playWithProfile(){
+     if (this._isPaused) {
+       this._resume();
+     } else if (!this._isPlaying) {
+       this._start();
+     }
+     var point1 = 0.8;
+     var point2 = 0.1;
+     var point3 = 0.1;
+     var point4 = 0.4;
+     
+     while (new Date().getTime() < this._startTime + this._buffer.duration*1000) {
+       var cpt = (new Date().getTime() - this._startTime) / (this._buffer.duration*1000) ;
```

Comment:
I'd taken a look at the `currentTime` property on `AudioContext`, but the problem seems to be that it is ever-increasing. So the moment the page is loaded, a counter starts going up, as far as I could tell. So I'm not sure if that would support a button press, that could occur at anytime, and then adjust the sound profile based off of the `currentTime`.

I'll take a look at `ex.Timer`